pydomcfg/utils.py

```diff
@@ -22,6 +22,56 @@ def _are_nemo_none(var: Iterable) -> Iterator[bool]:
         yield _is_nemo_none(v)
 
 
+# -----------------------------------------------------------------------------
+def _calc_rmax(depth: DataArray) -> float:
+    """
+    Calculate rmax: measure of steepness
+    This function returns the maximum slope paramater
+
+    rmax = abs(Hb - Ha) / (Ha + Hb)
+
+    where Ha and Hb are the depths of adjacent grid cells (Mellor et al 1998).
+
+    Reference:
+    *) Mellor, Oey & Ezer, J Atm. Oce. Tech. 15(5):1122-1131, 1998.
+
+    Parameters
+    ----------
+    depth: DataArray
+        Bottom depth (units: m).
+
+    Returns
+    -------
+    float
+        Maximum slope parameter (units: None)
+    """
+
+    # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    # Do we actually need this? mypy complains since
+    # DataArray.reset_indexe() returns Optional["DataArray"]
+    #
+    # depth = depth.reset_index(list(depth.dims))
```

Comment:
Interesting, not sure why they assigned that type annotation to `reset_index`. One quick solution is `DataArray(depth.reset_index(list(depth.dims)))`.

`_calc_rmax` doesn't really need this, the problem is that if we don't use it a DataArray with different indexes is returned. This means that we'll get conflicts when we try to merge in the main dataset. 

I was just lazy though... it's quite easy in this case to properly use the indexes. I'll open a separate PR for that as you haven't removed `calc_rmax` from `Bathymetry` yet.

Reply:
Ok, I will implement the quick solution `DataArray(depth.reset_index(list(depth.dims)))` for now i my branch since it seems we need it also in _calc_rmax: if I remove it, I get the following error: 
`ValueError: cannot reindex or align along dimension 'y' because the index has duplicate values`

Reply:
BTW, forza azzurri !!!!